### PR TITLE
CMake: Add binary paths for tests to the include paths

### PR DESCRIPTION
**Pull request by olesalscheider**
This PR contains three small fixes for the CMake build system.

### CODE REVIEW

**Comment by jlaine** [comment]
I don't understand all of the changes, but if we could confirm it fixes #141 I'm OK with you merging it.

It feels as though we need to set up appveyor to run automatic win32 builds for us.

**Comment by jlaine** [comment]
Actually wait I'm not able to reproduce the issue with building out of source directory. I'll cherry-pick c++-11 change but I don't understand the rest of the changes.

**Comment by jlaine** [comment]
> Why do you need this change at all?
> GCC use -std=gnu++14 by default since version 6.x.
> See: https://gcc.gnu.org/gcc-6/changes.html

Hm that explains why I'm not getting a warning locally, only on travis (Ubuntu/xenial). However gcc 4.8 is still a supported platform by Qt:

http://doc.qt.io/qt-5/supported-platforms.html

**Comment by olesalscheider** [comment]
Yes, the default standard is different for different versions of gcc. Also I'm not sure if CMake might set a default because I also get this warning with GCC 8.2.
So it is definitely better to be explicit about the standard that we want to use and support.

**Comment by olesalscheider** [comment]
> Actually wait I'm not able to reproduce the issue with building out of source directory. I'll cherry-pick c++-11 change but I don't understand the rest of the changes.

I don't have the issue on my computer but I could reproduce it on an Ubuntu 16.04 system. There it is fixed by the first commit in this PR.

The last commit is important for projects using qxmpp: Without it, you can only try to find qxmpp (`find_package(qxmpp)`) but you cannot specify a minimum version (e.g. `find_package(qxmpp 0.9.4)`).

**Comment by jlaine** [comment]
For the build failure : instead of blindly hoping it fixes something I'd rather we reproduce the problem with our travis setup (which uses Ubuntu/xenial!).

The second change seems like it's unrelated, split it out to its own PR, let CI run an we can merge it.

**Comment by olesalscheider** [comment]
Our current travis setup does not build out-of-source. I guess that's why the problem does not occur there. Maybe we should change that.

**Comment by jlaine** [comment]
I've added the c++-11 change to master.

I've also submitted PR #151 to see whether Travis reproduces the build error.

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,8 @@ mark_as_advanced(VERSION_MAJOR VERSION_MINOR VERSION_PATCH VERSION_STRING)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/modules")
 
+set(CMAKE_CXX_STANDARD 11)
```
Comment (0xd34df00d) [inline CMakeLists.txt]: You also probably want `CMAKE_CXX_STANDARD_REQUIRED` set to true (cf. https://cmake.org/cmake/help/v3.1/prop_tgt/CXX_STANDARD_REQUIRED.html#prop_tgt:CXX_STANDARD_REQUIRED).

**Comment by olesalscheider** [comment]
I have split the PR and submitted new PRs for the other changes.
This PR now only contains the fix for older versions of CMake.

**Comment by jlaine** [comment]
As this is specifically to support older cmake versions, is there a way we can add such old cmake versions to our CI build matrix? Otherwise it's bound to break again somewhere down the line.

**Comment by olesalscheider** [comment]
I'm not sure. We would have to use the native CMake version inside the CI instead of the one provided by travis.

It is easy to reproduce the issue with Gitlab CI because it allows to run the CI in an unmodified xenial docker container: https://gitlab.com/olesalscheider/qxmpp/-/jobs/102949969

But I guess we cannot select the docker container with travis. The only other thing I can think of is that there might be multiple versions of CMake in the travis container: The native from xenial and a newer one from travis. In this case we could maybe call the original version...

**Comment by olesalscheider** [comment]
I have fixed the issue in https://github.com/qxmpp-project/qxmpp/pull/155

**Pull request merged**